experimental/language-generation/csharp_dotnetcore/05.a.multi-turn-prompt-with-language-fallback/MultiLingualTemplateEngine.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using ActivityBuilder = Microsoft.Bot.Builder.Dialogs.Adaptive.Generators.ActivityGenerator;
+using System;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class MultiLingualTemplateEngineManager
+    {
+        public Dictionary<string, TemplateEngine> TemplateEnginesPerLocale { get; set; } = new Dictionary<string, TemplateEngine>();
+        private LanguagePolicy LangFallBackPolicy;
+
+        public MultiLingualTemplateEngineManager(Dictionary<string, List<string>> lgFilesPerLocale)
+        {
+            if (lgFilesPerLocale == null)
+            {
+                throw new ArgumentNullException(nameof(lgFilesPerLocale));
+            }
+
+            InternalConstructor(lgFilesPerLocale);
+        }
+
+        public Activity GenerateActivityForLocale(string templateName, object data, WaterfallStepContext stepContext)
```

Comment:
```suggestion
        public Activity GenerateActivityByLocale(string templateName, object data, WaterfallStepContext stepContext)
```

Reply:
Perhaps it's good to just call this GenerateActivity

Reply:
Another thing I would suggest is we put the "WaterfallStepContext" as first parameter, so that we can give data default to null. 

And the GenerateByLocal's first parameter is locale, very straightforward.

Reply:
Fixed. I left signature unchanged since I have overloads for all other types.